doc/reported_states_during_deployment.md

```diff
@@ -0,0 +1,353 @@
+set on the Iot Hub service and read by the IoT Edge. The second describes properties set by the IoT Edge
+and reported back to the service.
+
+## Set by IoT Hub service
+
+The following sections are set in \$edgeAgent twin by the IoT Hub service and are read by the IoT Edge 
+runtime:
+
+### Configuration section
+
+This section is present when the service applies at-scale (base and layered) deployments. This 
+section will not be present if no at-scale or layered deployment is applied.
+
+Example:
+
+```json
+{
+    "configurations" : {
+        "at-scale1": {
+            "status": "Applied"
+        },
+        "at-scale2": {
+            "status": "Targeted"
+        }
+    }
+}
+```
+
+| Relevant Field | Meaning |
+|------------------------------------------|-------------------------------------------------------|
+| .configurations.*\<deployment name\>*.status | "Applied" or "Targeted" for each deployment that may be targeted to this device. |
+
+
+### Desired Properties section
+
+This is the \$edgeAgent section of the modulesContent in an edge deployment.
+
+Example:
+
+```json
+{
+    "properties": {
+        "desired": {
+            "modules": {
+                "m1": {
+                    "settings": {
+                        "image": "mcr.microsoft.com/azureiotedge-simulated-temperature-sensor:1.0",
+                        "createOptions": ""
+                    },
+                    "type": "docker",
+                    "status": "running",
+                    "restartPolicy": "always",
+                    "version": "1.0"
+                }
+            },
+            "runtime": {
+                "settings": {
+                    "minDockerVersion": "v1.25"
+                },
+                "type": "docker"
+            },
+            "schemaVersion": "1.1",
+            "systemModules": {
+                "edgeAgent": {
+                    "settings": {
+                        "image": "mcr.microsoft.com/azureiotedge-agent:1.1",
+                        "createOptions": ""
+                    },
+                    "type": "docker"
+                },
+                "edgeHub": {
+                    "settings": {
+                        "image": "mcr.microsoft.com/azureiotedge-hub:1.1",
+                        "createOptions": "{\"HostConfig\":{\"PortBindings\":{\"443/tcp\":[{\"HostPort\":\"443\"}],\"5671/tcp\":[{\"HostPort\":\"5671\"}],\"8883/tcp\":[{\"HostPort\":\"8883\"}]}}}"
+                    },
+                    "type": "docker",
+                    "status": "running",
+                    "restartPolicy": "always"
+                }
+            },
+            "$version": 11
+        }
+    }
+}
+```
+ 
+| Relevant Field | Meaning |
+|------------------------------------------|-------------------------------------------------------|
+| .properties.desired."$version"           | This number is monotonically increasing. Even a rollback will increase this number. |
```

Comment:
Do we need to escape the `$` for version here?

Reply:
Going off the VS code render, no, but it doesn't hurt.